nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/wag/InvokeAWSGatewayApi.java

```diff
@@ -52,7 +51,6 @@
 import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 
```

Comment:
I don't understand why this isn't valid.  I put this there, if I recall because you may use this processor to read the data from an api to start a flow.  I thought this annotation was valid for that case.  Was I mistaken?

Reply:
Did a problem with this get reported?

Reply:
@ottobackwards the annotation is not needed for that. The @TriggerWhenEmpty annotation is only used if you want the processor to be triggered when there is an incoming connection and the incoming connection has no data. It's rarely needed. One example of where it is needed is MergeContent. In that case, there's a timeout that is configured, and so if you have 1,000 FlowFiles come in and get batched together, but no more data comes in for 24 hours, MergeContent should still be triggered so that it can timeout the 'bin' and create the merged FlowFile.

So in this case, consider a flow like GenerateFlowFile -> InvokeAWSGatewayAPI. And GenerateFlowFile runs every 30 minutes. InvokeAWSGatewayAPI is scheduled with Run Schedule of 0 seconds because it should run immediately and constantly when there is data. But because there is a `@TriggerWhenEmpty` annotation, InvokeAWSGatewayAPI is getting triggered constantly (even when there is no data queued up) and pounding the API service, which can be both inefficient and expensive.

So if there is no incoming connection, the processor will still get triggered even without the annotation.

Reply:
Ok, I miss understood the annotation then, sorry.

+1 fwiw.

Reply:
And thanks for the detailed explaination

Reply:
No worries, thanks @ottobackwards